docs/source/programming.md

```diff
@@ -0,0 +1,105 @@
+
+When multiple atoms are used, the state describing this multi-atom system is the tensor network of the states. Say these atoms are labeled $\[1, 2, ..., N\]$, then the state of the system is $\left|\Psi\right> = \left|\psi_1\right> \otimes \left|\psi_2\right> \otimes ... \otimes \left|\psi_N\right>$.
+
+2. Hamiltonian evolves the state
+
+In quantum physics, the state of a system of atoms evolve along time following the Schrödinger equation: $i\frac{d\left|\Psi\right>(t)}{dt} = \frac{H(t)}{\hbar} \left|\Psi\right>(t)$. $H(t)$ is the Hamiltonian describing the evolution of the system. **Pulser provides you the tools to program this Hamiltonian**. 
+
+2.1. The Hamiltonian
+
+The Hamiltonian describing the evolution of the system can be written as
+
+$$
+H(t) = \sum_i \left (H^D_i(t) + \sum_{j<i}H^\text{int}_{ij} \right),
+$$
+
+where $H^D_i$ is the driving Hamiltonian for atom $i$ and
+$H^\text{int}_{ij}$ is the interaction Hamiltonian between atoms $i$
+and $j$.
+
+2.1. Driving Hamiltonian
+
+The driving Hamiltonian describes the effect of a pulse of Rabi frequency $\Omega(t)$, detuning $\delta(t)$ and phase $\phi(t)$ on an individual atom, exciting the transition
+between two of its energies levels, $|a\rangle$ and $|b\rangle$.
+
+
+:::{figure} files/two_level_ab.png
+:align: center
+:alt: The energy levels for the driving Hamiltonian.
+:width: 200
+
+$$
+H^D(t) / \hbar = \frac{\Omega(t)}{2} e^{-i\phi(t)} |a\rangle\langle b| + \frac{\Omega(t)}{2} e^{i\phi(t)} |b\rangle\langle a| - \delta(t) |b\rangle\langle b|
+$$
+
+2.2. Interaction Hamiltonian
+
+The interaction Hamiltonian depends on the distance between the atoms $i$ and $j$, $R_{ij}$, and the energy levels in which the information is encoded in these atoms, that define the interaction between the atoms $\hat{E}_{ij}$
+
+$$
+H^\text{int}_{ij} = \hat{E}_{ij}(R_{ij})
+$$
+
+The interaction operator $\hat{E}_{ij}$ is composed of an entangling operator and an interaction strength:
+- If the rydberg state $\left|r\right>$ is involved in the computation, then $\hat{E}_{ij}(R_{ij}) = \frac{C_6}{R_{ij}^6} |r\rangle\langle r|_i |r\rangle\langle r|_j$, where the interaction strength is $\frac{C_6}{R_{ij}^6}$, with $C_6$ a coefficient that can be programmed using Pulser. The entangling operator between atom $i$ and $j$ is $\hat{n}_i\hat{n}_j = |r\rangle\langle r|_i |r\rangle\langle r|_j$.
+- If the information is stored in the $\left|0\right>$ and $\left|1\right>$ states of the `XY` basis, $\hat{E}_{ij}(R_{ij}) =\frac{C_3}{R_{ij}^3} (|1\rangle\langle 0|_i |0\rangle\langle 1|_j + |0\rangle\langle 1|_i |1\rangle\langle 0|_j)$, where the interaction strength is $\frac{C_3}{R_{ij}^3}$, with $C_3$ a coefficient that can be programmed using Pulser. The entangling operator between atom $i$ and $j$ is $\hat{\sigma}_i^{+}\hat{\sigma}_j^{-} + \hat{\sigma}_i^{-}\hat{\sigma}_j^{+} = |1\rangle\langle 0|_i |0\rangle\langle 1|_j + |0\rangle\langle 1|_i |1\rangle\langle 0|_j$.
+
```

Comment:
What about:
```suggestion
:::{important}
With Pulser, you program the interaction Hamiltonian by setting the distance between atoms, $R_{ij}$, and the choice of eigenstates used in the computation and the Rydberg level(s) targeted by the device. Most commonly, the ground and Rydberg states ($\left|g\right>$ and $\left|r\right>$) are used, such that
$$
H^\text{int}_{ij} = \frac{C_6}{R_{ij}^6} \hat{n}_i\hat{n}_j
$$
When providing the distance between atoms, Pulser ensures that you respect the constraints of your chosen device.
:::
```

Reply:
Just to be closer to the :::important section for the driving hamiltonian

Reply:
I'm ok with that, but the first sentence needs to be rephrased, I find it quite confusing. How about this:
```
With Pulser, you program the interaction Hamiltonian by setting the distance between atoms, $R_{ij}$. Additionally, the choice of eigenstates used in the computation and the Rydberg level(s) targeted by the device fully determine the interaction strength.
```

Reply:
With Pulser, you program the interaction Hamiltonian by setting the _relative_ distance between atoms, $R_{ij}$. 

The choice of the eigenstates used in the computation and therefore the Rydberg level(s) targeted is done by the user by selecting a specific device (link to section?) after inspecting its specifications. 

Most commonly, the ground and Rydberg states ($\left|g\right>$ and $\left|r\right>$) are used, such that

Reply:
I have added in the XY Hamiltonian that the |0> and |1> states are Rydberg states, because otherwise people would have gotten lost. @ferrulli1pasqal, I won't say _relative_ distance because it gives the impression that it's between 0 and 1 (relative to what ?) The chose of the eigenstates in the computation is not done by selecting the device, but by declaring the channels (and adding pulses to them, if you don't add any pulse it does not count).